src/singer_clojure/messages.clj

```diff
@@ -0,0 +1,83 @@
+(ns singer-clojure.messages
+  (:require [clojure.data.json :as json]
+            [clojure.test :refer [is]]))
+
+(defn valid?
```

Comment:
Should this be defn- (private)? I actually don't know. It's only used as the precondition in the write! method.

Reply:
We decided to leave these as is